CMakeLists.txt

```diff
@@ -523,6 +525,32 @@ if(OPENCV_DISABLE_FILESYSTEM_SUPPORT)
   add_definitions(-DOPENCV_HAVE_FILESYSTEM_SUPPORT=0)
 endif()
 
+if(CV_DISABLE_MULTI_THREAD)
+  add_definitions(-DCV_DISABLE_MULTI_THREAD)
+  # This is required to disable threads in the ts module, as
+  # described in `ts_gtest.h`.
+  add_definitions(-DGTEST_HAS_PTHREAD=0)
```

Comment:
I believe root CMake file should not know details of `ts` module.
Please move this into `modules/ts/CMakeLists.txt`, use

```
ocv_target_compile_definitions(${the_module} PUBLIC GTEST_HAS_PTHREAD=0)
```

instead.

Reply:
Done